nowledge-mem-copilot-cli-plugin/scripts/copilot-stop-save.py

```
@@ -0,0 +1,626 @@
+
+        if not text:
+            continue
+        if messages and messages[-1]["role"] == role:
+            messages[-1]["content"] += "\n\n" + text
+        else:
+            messages.append({"role": role, "content": text})
+    return messages
+
+
+def run_json(args: list[str]) -> dict:
+    proc = subprocess.run(args, capture_output=True, text=True)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)